test/src/main/java/io/zeebe/test/exporter/ExporterIntegrationRule.java

```diff
@@ -0,0 +1,327 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.zeebe.test.exporter;
+
+import static io.zeebe.test.util.record.RecordingExporter.workflowInstanceRecords;
+
+import com.moandjiezana.toml.Toml;
+import com.moandjiezana.toml.TomlWriter;
+import io.zeebe.broker.system.configuration.BrokerCfg;
+import io.zeebe.broker.system.configuration.ExporterCfg;
+import io.zeebe.client.ClientProperties;
+import io.zeebe.client.api.subscription.JobHandler;
+import io.zeebe.client.api.subscription.JobWorker;
+import io.zeebe.exporter.record.Record;
+import io.zeebe.exporter.record.value.IncidentRecordValue;
+import io.zeebe.exporter.spi.Exporter;
+import io.zeebe.model.bpmn.Bpmn;
+import io.zeebe.model.bpmn.BpmnModelInstance;
+import io.zeebe.protocol.intent.IncidentIntent;
+import io.zeebe.protocol.intent.WorkflowInstanceIntent;
+import io.zeebe.test.ClientRule;
+import io.zeebe.test.EmbeddedBrokerRule;
+import io.zeebe.test.util.TestUtil;
+import io.zeebe.test.util.record.RecordingExporter;
+import java.io.InputStream;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Consumer;
+import org.junit.rules.ExternalResource;
+
+/**
+ * JUnit test rule to facilitate running integration tests for exporters.
+ *
+ * <p>Sets up an embedded broker, gateway, client, and provides convenience methods to:
+ *
+ * <ol>
+ *   <li>run a sample workload (e.g. deploy & create workflows, setup a job worker, etc.)
+ *   <li>visit all exported records so far
+ *   <li>perform simple operations such as deployments, starting a job worker, etc.
+ * </ol>
+ *
+ * An example integration test suite could look like:
+ *
+ * <pre>
+ * public class MyExporterIT {
+ *   @Rule public final ExporterIntegrationRule exporterIntegrationRule = new ExporterIntegrationRule();
```

Comment:
We should mention that you could configure your exporter in `zeebe.test.cfg.toml` and the broker is started then automatically. Or that you can configure it in the directly on the Rule. Maybe two code examples, one with the minimal default use case. And one for the case where you want to use different configurations per case.